File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.util.*;
import java.io.*;

public class Main {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		Set<Integer> set1 = new HashSet<>();
		StringTokenizer st = new StringTokenizer(br.readLine());
		for(int i = 0; i < N; i++) {
			set1.add(Integer.parseInt(st.nextToken()));
		}
		
		StringBuilder sb = new StringBuilder();
		
		int M = Integer.parseInt(br.readLine());
		st = new StringTokenizer(br.readLine());
		for(int i = 0; i < M; i++) {
			if(set1.contains(Integer.parseInt(st.nextToken()))) {
				sb.append("1\n");
			}else {
				sb.append("0\n");
			}
		}
		

		System.out.println(sb);
	}

}

File: 백준/Silver/1920. 수 찾기/수 찾기.kt
package org.example

import java.util.StringTokenizer

fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.`out`.bufferedWriter()
    val sb = StringBuilder()
    val N = br.readLine().toInt()
    val stN = StringTokenizer(br.readLine())
    val hashSet = HashSet<String>()
    repeat(N) {
        hashSet.add(stN.nextToken())
    }
    val M = br.readLine().toInt()
    val stM = StringTokenizer(br.readLine())
    repeat(M) {
        if (hashSet.contains(stM.nextToken())) sb.append(1).append("\n")
        else sb.append(0).append("\n")
    }
    bw.write(sb.toString())
    bw.flush()
    bw.close()
    br.close()
}